### PR TITLE
sg: add "grafana.folder" argument to monitoring

### PR DESCRIPTION
**Pull request by danieldides**
Add optional `grafana.folder` argument to `sg monitoring generate` to allow the Grafana Client to upload the dashboards to a specific folder. `sg` will attempt to find the folder on the remote server and override the dashboards. If it can't find it, it will create it. `sg` will also set the UID of the resulting dashboards to include the folder name to keep them globally unique.

## Test plan

`sg start monitoring` to start a local Grafana instance. Verify that the only folder that exists is `General`.

Screen Shot 2022-10-04 at 16 28 40

### Loading dashboards into a non-existent folder

```bash

#!/bin/bash

SRC_LOG_LEVEL=debug go run ./dev/sg monitoring generate \
	--reload \
	--grafana.folder="test1" 
```

Look for debug messages like:
```
1665090245235709000	DEBUG	generate.grafana.client	monitoring/generator.go:125	github.com/sourcegraph/sourcegraph/monitoring/monitoring.Generate	No existing folder found, creating a new one	{"Resource": {"service.name": "sg", "service.version": "dev", "service.instance.id": "1e6cf8e8-f058-4f5a-9c88-6d1317d0b51c"}, "Attributes": {}}
```

And verify the folder now exists and the dashboards are populated correctly:
Screen Shot 2022-10-06 at 16 04 47

### Loading dashboard into pre-existing folder

```bash

#!/bin/bash

SRC_LOG_LEVEL=debug go run ./dev/sg monitoring generate \
	--reload \
	--grafana.folder="test1" 
```

and look for log messages like:
```
1664919020647296000	DEBUG	generate.grafana	monitoring/generator.go:134	github.com/sourcegraph/sourcegraph/monitoring/monitoring.Generate	Existing folder found. Using it.	{"Resource": {"service.name": "sg", "service.version": "dev", "service.instance.id": "70a705bd-80ac-4ffb-b94e-ad151290c268"}, "Attributes": {"dashboard": "frontend", "instance": "http://127.0.0.1:3370"}}
```

and verify the dashboards still exist.

### CODE REVIEW

**Comment by bobheadxi** [comment]
I added https://github.com/sourcegraph/sourcegraph/pull/42545 on top of this which makes sure everything plays nice, at least locally:

- remove volume mount for disk dashboards in `sg run grafana`, which prevents Grafana reloading from disk and resetting changes
- add a reload for the Home dashboard

However, I think we need to be able to set the `UID` on each board to be specific to a folder - looks like they really do need to be unique

Note: I tried to marshal the home json into `*sdk.Board` and it doesn't work

File: sg.config.yaml [hunk 1]
```diff
@@ -593,7 +593,6 @@ commands:
         -p 0.0.0.0:3370:3370 ${ADD_HOST_FLAG} \
         -v "${GRAFANA_DISK}":/var/lib/grafana \
         -v "$(pwd)"/dev/grafana/all:/sg_config_grafana/provisioning/datasources \
-        -v "$(pwd)"/docker-images/grafana/config/provisioning/dashboards:/sg_grafana_additional_dashboards \
```
Comment (danieldides) [inline sg.config.yaml:596]: @bobheadxi I believe you made this change, is it intended to be included?

Comment (bobheadxi) [reply sg.config.yaml:596]: Yes, it prevents the hot-reloading from working correctly because Grafana _also_ hot-reloads itself from disk it seems

Comment (bobheadxi) [reply sg.config.yaml:596]: we could add an env config to add this volume mount back or exclude it

**Review by bobheadxi — APPROVED** [review]
LGTM - as the complexity of this grows, I think we can introduce a new command at some point that can be more granularly tailored towards generating dashboards, e.g.

```
sg monitoring generate dashboards
```

This will free us from a lot of the baggage that is currently in `Generate`

**Pull request merged**